src/Build/Utilities/EngineFileUtilities.cs

```diff
@@ -332,7 +332,27 @@ private static string[] GetFileList(
                     // as a relative path, we will get back a bunch of relative paths.
                     // If the filespec started out as an absolute path, we will get
                     // back a bunch of absolute paths
-                    (fileList, _, _) = fileMatcher.GetFiles(directoryUnescaped, filespecUnescaped, excludeSpecsUnescaped);
+                    (fileList, _, _, string globFailure) = fileMatcher.GetFiles(directoryUnescaped, filespecUnescaped, excludeSpecsUnescaped);
```

Comment:
It looks concerning to have this tuple growing.
Should we wrap it with readable struct? @JanKrivanek ?

Reply:
We could also get rid of the 2nd and 3rd parameter altogether from the internal API. FileMatcher is `internal` and they're not used anywhere. Any reason not to do that?

Reply:
There in fact is a reason: tests use them and there is no easy way to refactor if we want to test inner behavior of file matching, we need an `internal` method returning this information. Unsure if struct would be helpful.

Reply:
Since this is a private function and it is used on testing I think we can let this be as it is right and and maybe make an issue to improve / refactor this area with the tests in mind.